### DISCUSSION TITLE
The recall of FLAT and HNSW is not as expected

### DISCUSSION THREAD

**Discussion by CharlesWu123** [milvus-io/milvus#34671] [Q&A and General discussion]
Using the same question and threshold for querying, normally flat's recall rate should be higher than hnsw, but my test results showed that flat's recall rate was lower than hnsw. Here are my parameters, all of which use iterators to return all the content that meets the criteria:
flat：
search_config: 
```json
{
"metric_type": "COSINE",
"params": {
  "radius": 0.6,
  "range_filter": 1.0
  }
}
```
hnsw:
index_config:
```json
{
"index_type": "HNSW",
"params": {
  "M": 8,
  "efConstruction": 128
  },
"metric_type": "COSINE"
}
```
search_config:
```json
{
"params": {
  "ef": 10000,
  "radius": 0.6,
  "range_filter": 1.0
  },
"metric_type": "COSINE"
}
```
Using the above conditions and the same question, flat retrieved 7615 results and hnsw retrieved 8751 results, resulting in 6122 identical results.
Is this result normal?

**Comment by liliu-z** [comment]
This is not as expected. Do you mind to help take a look @alwayslove2013

**Comment by liliu-z** [comment]
@CharlesWu123 Is this still a problem? I noticed you closed this

**Reply by CharlesWu123** [reply]
This is my problem. I accidentally tested on two separate collection twice, and the data was slightly different.

**Reply by liliu-z** [reply]
@CharlesWu123 Thanks for this clarification
